### PR TITLE
Added ability to set user password when adding a new user

### PR DESCRIPTION
**Pull request by blakeaowens**
The ability to change password already exists, however this removes the previously used "generate unusable password" feature when creating a new user.

### CODE REVIEW

File: dojo/user/views.py [hunk 1]
```diff
@@ -284,7 +284,7 @@ def add_user(request):
         contact_form = UserContactInfoForm(request.POST)
         if form.is_valid() and contact_form.is_valid():
             user = form.save(commit=False)
-            user.set_unusable_password()
+            user.set_password(request.POST['password'])
```
Comment (valentijnscholten) [inline dojo/user/views.py:287]: not sure if this is what we want / users expect. unusable password can be useful. maybe if left empty, set unusable? or have checkbox/radiobutton? or settings option?

**Review by valentijnscholten — CHANGES_REQUESTED** [review]
see above

**Comment by valentijnscholten** [comment]
Thanks for the PR, there was some talk about this to have this added to the API. Would you be willing to look at adding it there as well?

File: dojo/user/views.py (hunk 1 above)
Comment (blakeaowens) [reply dojo/user/views.py:287]: The goal of this PR was to get rid of the admin page. From my understanding, "set_unusable_password()" was used initially as a temporary solution when creating users.

Comment (blakeaowens) [reply dojo/user/views.py:287]: Also, maybe adding this to the API should be done in a different PR.

Comment (valentijnscholten) [reply dojo/user/views.py:287]: Could you describe what problem you are trying to solve?

Comment (blakeaowens) [reply dojo/user/views.py:287]: The problem I am solving is getting rid of the admin page. Unusable passwords can only be reset through the admin page. With that page removed, there needs to be a way to set an initial password for a new user that can be changed through the UI, which cannot be done with unusable passwords.

**Review by valentijnscholten — CHANGES_REQUESTED** [review]
As said above I think the operator should be able to choose what to do. I am using LDAP and when creating users I do actually want to have the unusable password as I don't want users to start resetting their passwords. 

The Django Admin is currently used to assign model permissions to users, for example when creating CI/CD users with least privilege. So we can't disable the admin yet as long as we are not completely migrated to the new Authorization V2.

**Comment by Maffooch** [comment]
I did not think the django model permissions have ever worked with DefectDojo. This was because when dojo was originally written, the Django permissions did not sufficiently cover the authorized user permission scheme currently in place.

Also looks like the admin page is not removed, but just disabled by default. Before, since the admin page enabled variable was set as an environment variable through the docker file, there was no way to disable it without a rebuild. Now it looks like the admin page can be enabled/disabled through settings as it was intended.

**Comment by valentijnscholten** [comment]
The permissions are working, @madchap is using them and we've been instructing others on slack to use them.

**Comment by Maffooch** [comment]
Oh interesting! I wish I would've none that years ago. Either way, the admin page is still available. Just needs to be enabled in settings first.

**Comment by valentijnscholten** [comment]
Yeah, easiest would be to leave it enabled until we have the new authorization V2. But if other insist on disabling it by default, we should add some docs to re-enable it easily.

File: dojo/user/views.py (hunk 1 above)
Comment (blakeaowens) [reply dojo/user/views.py]: With my latest commit the Django Admin page is enabled in settings (can easily be disabled), and unusable password functionality is still available. When creating a new user, an unusable password will be set if the password field is left blank (added help text to this field on the add user page for clarification). I also tagged on a fix for a problem I discovered with the change password logic, which previously allowed the user to successfully leave both password fields blank.

**Review by Maffooch — APPROVED** [review]
Looks ready to go now

**Review by valentijnscholten — APPROVED** [review]

**Comment by valentijnscholten** [comment]
Good solution, thanks.

**Pull request merged**